pkg/spoke/managedcluster/healthcheck_controller.go

```diff
@@ -0,0 +1,90 @@
+package managedcluster
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	clientset "github.com/open-cluster-management/api/client/cluster/clientset/versioned"
+	clusterv1informer "github.com/open-cluster-management/api/client/cluster/informers/externalversions/cluster/v1"
+	clusterv1listers "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1"
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	"github.com/open-cluster-management/registration/pkg/helpers"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	discovery "k8s.io/client-go/discovery"
+)
+
+// managedClusterHealthCheckController checks the kube-apiserver health on managed cluster to determine it whether is available.
+type managedClusterHealthCheckController struct {
+	clusterName      string
+	hubClusterClient clientset.Interface
+	hubClusterLister clusterv1listers.ManagedClusterLister
+	discoveryClient  discovery.DiscoveryInterface
+}
+
+// NewManagedClusterHealthCheckController creates a managed cluster health check controller on managed cluster.
+func NewManagedClusterHealthCheckController(
+	clusterName string,
+	hubClusterClient clientset.Interface,
+	hubClusterInformer clusterv1informer.ManagedClusterInformer,
+	discoveryClient discovery.DiscoveryInterface,
+	recorder events.Recorder) factory.Controller {
+	c := &managedClusterHealthCheckController{
+		clusterName:      clusterName,
+		hubClusterClient: hubClusterClient,
+		hubClusterLister: hubClusterInformer.Lister(),
+		discoveryClient:  discoveryClient,
+	}
+
+	return factory.New().
+		WithInformers(hubClusterInformer.Informer()).
+		WithSync(c.sync).
+		ResyncEvery(1*time.Minute).
```

Comment:
I think you need this resync configurable based on check frequency, right?

Reply:
make it configurable